.github/workflows/unit-test.yml

```diff
@@ -22,6 +22,7 @@ jobs:
       working-directory: ./build
       run: xvfb-run --auto-servernum --server-args "-screen 0 1024x768x24" ant test -Drunning-from-github-action=1
     - name: Publish results
+      if: always()
```

Comment:
What does this do exactly? Does `if` default to something like "all previous steps were successful" and this overrides to always run? Maybe this could use a comment in that case?

Reply:
If the previous step fails (i.e. the Unit Test step) then this step (i.e. the Publishing Step) is not executed (for example see https://github.com/cmaglie/Arduino/runs/664274907?check_suite_focus=true)
Before this patch the CI published the results only if all test passes... that's not very informative :-)